### PR TITLE
Sync Halo master and resolve Strix integration conflicts

### PR DESCRIPTION
**Pull request by LaurentZuijdwijk**
## Overview

Sync `halo-box/llama.cpp:master` at `29e091ea5b228ac1735cde369e68e6767a53e510` into the Strix fork, resolving the conflicts from #43 on a dedicated Strix branch. This replaces #43 without adding Strix-specific changes to the parent repository.

The merge starts from Strix `6b1f3082bd83dbbdb4cb62c47aca3af2e4daf733` and resolves conflicts in 11 files. It preserves Strix's scheduler split limit, AMD attention configuration, direct top-k attention, reasoning-budget controls and custom quantization paths. Qwen 3.5 now uses the upstream QKV builder while retaining the exact-batch hint through its projection views. Both sets of RMS normalization regression cases are retained.

### Vulkan integration choices

- Keep Strix's existing matrix pipeline selection, tuning and ROCmFPx shader variants. The Vulkan portions of upstream commits `1945e092` (Intel f16-B/tile changes), `91f6a6cf` (matrix A-type specialization refactor) and `6788edb4` (dependent small-M dispatch changes) are deliberately deferred. They need a separate port that preserves Strix's custom paths; this merge does not claim to integrate those implementations.
- Keep Strix's existing DeepSeek hyper-connection shaders and matching host interfaces instead of duplicating upstream `7a333e72`.
- Integrate the remaining Vulkan updates, including argsort fixes, TQ1_0 support, type-aligned GET_ROWS, RMS normalization fusion and unary fusion. Remove the superseded duplicate Strix unary-fusion path.

## Validation

Status: **INCOMPLETE / GPU runtime unverified**. The user explicitly requested no GPU use.

- Combined-tree Release CPU build, tools/server and embedded UI build: passed.
- CPU-only main CTest selection: 54/54 passed, including recurrent rollback, state save/load and scheduler tests. External tokenizer-repository download and model-dependent eval-callback were excluded.
- Focused CPU operation checks for RMS_NORM_MUL, RMS_NORM_MUL_ADD, RMS_NORM_MUL_ROPE and DSV4_HC_COMB/PRE/POST: 280/280 passed. These do not validate GPU execution.
- Vulkan backend and shader build: passed. Final Vulkan source syntax check: passed (compile only, no device execution).
- Conflict-marker and staged whitespace checks: passed.

No GPU workloads or hardware performance benchmarks were run. Vulkan runtime, ROCm/CUDA builds and execution, Metal/OpenCL/Hexagon backends, real-model output equivalence and perplexity remain unverified. No performance claim is made.

## Requirements

- Repository contributing guidelines reviewed by Codex.
- Scope: user-requested synchronization from the parent fork, retaining Strix-specific behavior where implementations conflict.
- AI usage disclosure: AGENT-AUTHORED conflict resolution, merge commit and PR description by Codex at the user's request. The commit includes an `Assisted-by: Codex` trailer.
- What was NOT verified: GPU/runtime/hardware and real-model coverage listed above. Keep this PR in draft until the pending integration review and GPU validation are complete.

### CODE REVIEW

**Comment by SteelPh0enix** [comment]
I've tested this on my Framework Desktop; built via Nix for Vulkan, ran https://huggingface.co/agentionai/Signal-3.8-Flash-Next-GGUF with MTP and mmproj enabled.

Seems to work just fine:

```log
[1:19:05 PM] wrz 12 13:08:50 RX-78-FPC llm-router[5098]: 1.47.930.960 I srv  proxy_reques: proxying request to model qwen-next on port 37913
[1:19:05 PM] wrz 12 13:08:50 RX-78-FPC llm-router[5098]: [37913] 0.33.369.439 I slot get_availabl: id  0 | task -1 | selected slot by LRU, t_last = 300524724
[1:19:05 PM] wrz 12 13:08:50 RX-78-FPC llm-router[5098]: [37913] 0.33.410.131 W common_sampler_init: backend sampling is not compatible with grammar, disabling
[1:19:05 PM] wrz 12 13:08:50 RX-78-FPC llm-router[5098]: [37913] 0.33.410.157 I slot launch_slot_: id  0 | task 35 | processing task, is_child = 0
[1:19:05 PM] wrz 12 13:09:07 RX-78-FPC llm-router[5098]: [37913] 0.50.201.622 I slot print_timing: id  0 | task 35 | prompt processing, n_tokens =   8194, progress = 0.39, t =  14.63 s / 559.95 tokens per second
[1:19:05 PM] wrz 12 13:09:26 RX-78-FPC llm-router[5098]: [37913] 1.08.763.404 I slot print_timing: id  0 | task 35 | prompt processing, n_tokens =  16386, progress = 0.75, t =  32.97 s / 497.06 tokens per second
[1:19:05 PM] wrz 12 13:09:34 RX-78-FPC llm-router[5098]: [37913] 1.17.549.890 I slot print_timing: id  0 | task 35 | prompt processing, n_tokens =  19852, progress = 0.91, t =  42.24 s / 469.99 tokens per second
[1:19:05 PM] wrz 12 13:09:40 RX-78-FPC llm-router[5098]: [37913] 1.22.666.058 I slot print_timing: id  0 | task 35 | prompt processing, n_tokens =  21900, progress = 1.00, t =  47.59 s / 460.14 tokens per second
[1:19:05 PM] wrz 12 13:09:43 RX-78-FPC llm-router[5098]: [37913] 1.26.011.870 I slot print_timing: id  0 | task 35 | n_gen =    103, tg =  32.35 t/s, tg_3s =  32.66 t/s
[1:19:05 PM] wrz 12 13:09:46 RX-78-FPC llm-router[5098]: [37913] 1.29.081.973 I slot print_timing: id  0 | task 35 | n_gen =    209, tg =  33.42 t/s, tg_3s =  34.53 t/s
[1:19:05 PM] wrz 12 13:09:49 RX-78-FPC llm-router[5098]: [37913] 1.32.157.406 I slot print_timing: id  0 | task 35 | n_gen =    334, tg =  35.81 t/s, tg_3s =  40.64 t/s
[1:19:05 PM] wrz 12 13:09:52 RX-78-FPC llm-router[5098]: [37913] 1.35.180.857 I slot print_timing: id  0 | task 35 | n_gen =    435, tg =  35.22 t/s, tg_3s =  33.41 t/s
[1:19:05 PM] wrz 12 13:09:55 RX-78-FPC llm-router[5098]: [37913] 1.38.224.743 I slot print_timing: id  0 | task 35 | n_gen =    534, tg =  34.69 t/s, tg_3s =  32.52 t/s
[1:19:05 PM] wrz 12 13:09:58 RX-78-FPC llm-router[5098]: [37913] 1.41.256.955 I slot print_timing: id  0 | task 35 | n_gen =    651, tg =  35.33 t/s, tg_3s =  38.59 t/s
[1:19:05 PM] wrz 12 13:10:01 RX-78-FPC llm-router[5098]: [37913] 1.44.298.332 I slot print_timing: id  0 | task 35 | n_gen =    751, tg =  34.98 t/s, tg_3s =  32.88 t/s
[1:19:05 PM] wrz 12 13:10:04 RX-78-FPC llm-router[5098]: [37913] 1.47.347.308 I slot print_timing: id  0 | task 35 | n_gen =    851, tg =  34.71 t/s, tg_3s =  32.80 t/s
[1:19:05 PM] wrz 12 13:10:07 RX-78-FPC llm-router[5098]: [37913] 1.50.359.990 I slot print_timing: id  0 | task 35 | n_gen =    948, tg =  34.43 t/s, tg_3s =  32.20 t/s
[1:19:05 PM] wrz 12 13:10:09 RX-78-FPC llm-router[5098]: [37913] 1.51.765.512 I slot print_timing: id  0 | task 35 | prompt eval time =   49448.31 ms / 21904 tokens (    2.26 ms per token,   442.97 tokens per second)
[1:19:05 PM] wrz 12 13:10:09 RX-78-FPC llm-router[5098]: [37913] 1.51.765.515 I slot print_timing: id  0 | task 35 |        eval time =   28906.70 ms /   993 tokens (   29.14 ms per token,    34.32 tokens per second)
[1:19:05 PM] wrz 12 13:10:09 RX-78-FPC llm-router[5098]: [37913] 1.51.765.516 I slot print_timing: id  0 | task 35 |       total time =   78355.01 ms / 22897 tokens
[1:19:05 PM] wrz 12 13:10:09 RX-78-FPC llm-router[5098]: [37913] 1.51.765.517 I slot print_timing: id  0 | task 35 |    graphs reused =        136
[1:19:05 PM] wrz 12 13:10:09 RX-78-FPC llm-router[5098]: [37913] 1.51.765.528 I slot print_timing: id  0 | task 35 | draft acceptance = 0.87739 (  551 accepted /   628 generated), mean len =  2.90
[1:19:05 PM] wrz 12 13:10:09 RX-78-FPC llm-router[5098]: [37913] 1.51.765.910 I slot      release: id  0 | task 35 | stop processing: n_tokens = 23316, truncated = 0
```

Image recognition works fine too.

**Comment by gaetan-puleo** [comment]
Qwen 3.5 correctness  :

- Not same logits but linked to new QKV cache for Qwen35

Ok for me

**Review by gaetan-puleo — APPROVED** [review]
